doc/doxygen/code-gallery/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2015 by the deal.II Authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+
+
+#
+# A target for the preparation of all the stuff happening in here...
+#
+
+ADD_CUSTOM_TARGET(code-gallery)
+
+
+#
+# Check whether someone has either specified a global variable that
+# points to a code gallery directory, or whether it has been put into
+# a top-level code-gallery/ directory alongside the tutorial/
+# directory
+#
+SET_IF_EMPTY(DEAL_II_CODE_GALLERY_DIRECTORY ${CMAKE_SOURCE_DIR}/code-gallery)
+IF (IS_DIRECTORY ${DEAL_II_CODE_GALLERY_DIRECTORY})
```

Comment:
It is usually better to use a conditional
```
IF (EXISTS ${DEAL_II_CODE_GALLERY_DIRECTORY}/CMakeLists.txt)
```
This also tests that the directory is not empty and might contain what we are actually looking for.